central/complianceoperator/v2/scanconfigurations/service/service_impl.go

```diff
@@ -176,8 +181,38 @@ func (s *serviceImpl) DeleteComplianceScanConfiguration(ctx context.Context, req
 	if req.GetId() == "" {
 		return nil, errors.Wrap(errox.InvalidArgs, "Scan configuration ID is required for deletion")
 	}
+	// Snapshots get deleted with the ScanConfiguration we need to delete the BlobData before
```

Comment:
So, are we deleting here because it is impossible to download the report after deletion, or because we want to prevent it on purpose?

Both are fine as long as we inform the user that they loose access to past reports (that maybe a task for the UI).

Reply:
Snapshots are deleted automatically when the scan configuration is deleted. This is because they become inaccessible. Same happens with the downloads although not automatically so we need to delete them here.